panel/widgets/chat.py

```diff
@@ -0,0 +1,1085 @@
+        The size of each icon.
+    active_icons : Dict
+        The mapping of reactions to their corresponding active icon names;
+        if not set, the active icon name will default to its "filled" version.
+    """
+
+    value = param.List(doc="The selected reactions.")
+
+    options = param.Dict(default={"favorite": "heart"}, doc="""
+        A key-value pair of reaction values and their corresponding tabler icon names
+        found on https://tabler-icons.io.""")
+
+    icon_size = param.String(default="15px", doc="""The size of each icon.""")
+
+    active_icons = param.Dict(default={}, doc="""
+        The mapping of reactions to their corresponding active icon names;
+        if not set, the active icon name will default to its "filled" version.""")
+
+    _icon_base_url = param.String("https://tabler-icons.io/static/tabler-icons/icons/")
+
+    _stylesheets = [f"{CDN_DIST}css/chat_reaction_icons.css"]
+
+    _template = """
+        <div id="reaction-icons" class="reaction-icons">
+            {% for reaction, icon_name in options.items() %}
+            <img id="icon-{{ loop.index0 }}" alt="{{ reaction }}"
+                {% if reaction in value and reaction in active_icons %}
+                src="${_icon_base_url}{{ active_icons[reaction] }}.svg"
+                {% elif reaction in value %}
+                src="${_icon_base_url}{{ icon_name }}-filled.svg"
+                {% else %}
+                src="${_icon_base_url}{{ icon_name }}.svg"
+                {% endif %}
+                style="width: ${icon_size}; height: ${icon_size}; cursor: pointer;"
+                onclick="${script('update_value')}">
+            </img>
+            {% endfor %}
+        </div>
+    """
+
+    _scripts = {
+        "update_value": """
+            const reaction = event.target.alt;
+            const icon_name = data.options[reaction];
+            if (data.value.includes(reaction)) {
+                data.value = data.value.filter(r => r !== reaction);
+                event.target.src = data._icon_base_url + icon_name + ".svg";
+            } else {
+                data.value = [...data.value, reaction];
+                if (reaction in data.active_icons) {
+                    event.target.src = data._icon_base_url + data.active_icons[reaction] + ".svg";
+                } else {
+                    event.target.src = data._icon_base_url + icon_name + "-filled.svg";
+                }
+            }
+        """
+    }
+
+    _stylesheets: ClassVar[List[str]] = [
+        f"{CDN_DIST}css/chat_reaction_icons.css"
+    ]
+
+    def __init__(self, **params):
+        super().__init__(**params)
+        if self.icon_size.endswith("px"):
+            self.max_width = int(self.icon_size[:-2])
+
+
+class ChatEntry(CompositeWidget):
+    """
+    A widget to display a chat entry.
+
+    Parameters
+    ----------
+    value : object
+        The message contents. Can be a string, pane, widget, layout, etc.
+    user : str
```

Comment:
Why not use the `name` parameter for this value?

I think it can be a bit confusing that the parameter name is `user` and its value is often 'user'.

Reply:
I suspect if named `name` it might also have issues like:
https://github.com/holoviz/panel/issues/5377#issue-1841704907